openapiart/common.py

```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
 
 class OpenApiValidator(object):
 
-    _MAC_REGEX = re.compile(r"^([\da-fA-F]{2}[:]){5}[\da-fA-F]{2}$")
-    _IPV6_REP1 = re.compile(r"^:[\da-fA-F].+")
-    _IPV6_REP2 = re.compile(r".+[\da-fA-F]:$")
-    _IPV6_REP3 = re.compile(r"^" + r"[\da-fA-F]{1,4}:" * 7 + r"[\da-fA-F]{1,4}$")
-    _HEX_REGEX = re.compile(r"^0?x?[\da-fA-F]+$")
-
     __slots__ = ()
 
     def __init__(self):
         pass
 
     def validate_mac(self, mac):
-        if mac is None:
+        if mac is None or not isinstance(mac, str):
+            return False
+        try:
+            if len(mac) != 17:
+                return False
+            return all([
+                0 <= int(oct, 16) <= 255 for oct in mac.split(":")
+            ])
+        except Exception:
             return False
-        if isinstance(mac, list):
-            return all([True if self._MAC_REGEX.match(m) else False for m in mac])
-        if self._MAC_REGEX.match(mac):
-            return True
-        return False
 
     def validate_ipv4(self, ip):
-        if ip is None:
+        if ip is None or not isinstance(ip, str):
+            return False
+        if len(ip.split(".")) != 4:
             return False
         try:
-            if isinstance(ip, list):
-                return all([all([0 <= int(oct) <= 255 for oct in i.split(".", 3)]) for i in ip])
-            else:
-                return all([0 <= int(oct) <= 255 for oct in ip.split(".", 3)])
+            return all([0 <= int(oct) <= 255 for oct in ip.split(".", 3)])
         except Exception:
             return False
 
-    def _validate_ipv6(self, ip):
-        if ip is None:
-            return False
-        if self._IPV6_REP1.match(ip) or self._IPV6_REP2.match(ip):
+    def validate_ipv6(self, ip):
+        if ip is None or not isinstance(ip, str):
```

Comment:
Not showing any error for trailing whitespaces
```python
ip.gateway = '2a00:1450:f013:c03:0:0:0:1 '
```

Reply:
I feel trailing spaces can be striped, we may need to test with illegal char or space with in the address. will check that part